File: src/difficulty_1/difficulty_1_4/zamka/Zamka.java
package difficulty_1.difficulty_1_4.zamka;

import java.util.Scanner;

public class Zamka {

    public static void main (String[] args) {
        Scanner scanner = new Scanner(System.in);

        int L = scanner.nextInt();
        int D = scanner.nextInt();
        int X = scanner.nextInt();

        for(int i = L; i <= D; i++) {
            int sum = String.valueOf(i).chars()
                    .map(Character::getNumericValue)
                    .sum();
            if(sum == X) {
                System.out.println(i);
                break;
            }
        }

        for(int i = D; i >= L; i--) {
            int sum = String.valueOf(i).chars()
                    .map(Character::getNumericValue)
                    .sum();
            if(sum == X) {
                System.out.println(i);
                break;
            }
        }

    }
}

File: KotlinImplementation/src/difficulty_1/Zamka.kt
package difficulty_1

fun main() {
    val L = readLine()!!.toInt()
    val D = readLine()!!.toInt()
    val X = readLine()!!.toInt()

    for (i in L..D) {
        val sum = i.toString().chars()
            .map { codePoint: Int -> Character.getNumericValue(codePoint) }
            .sum()
        if (sum == X) {
            println(i)
            break
        }
    }

    for (i in D downTo L) {
        val sum = i.toString().chars()
            .map { codePoint: Int -> Character.getNumericValue(codePoint) }
            .sum()
        if (sum == X) {
            println(i)
            break
        }
    }
}